umbral/keys.py

```diff
@@ -108,6 +111,43 @@ def get_pubkey(self):
         """
         return UmbralPublicKey(self.bn_key * self.params.g)
 
+    def to_cryptography_privkey(self):
+        """
+        Returns a cryptography.io EllipticCurvePrivateKey from the Umbral key.
+        """
+        backend = default_backend()
+
+        backend.openssl_assert(self.bn_key.group != backend._ffi.NULL)
+        backend.openssl_assert(self.bn_key.bignum != backend._ffi.NULL)
+
+        ec_key = backend._lib.EC_KEY_new()
+        backend.openssl_assert(ec_key != backend._ffi.NULL)
+        ec_key = backend._ffi.gc(ec_key, backend._lib.EC_KEY_free)
+
+        res = backend._lib.EC_KEY_set_group(ec_key, self.bn_key.group)
+        backend.openssl_assert(res == 1)
+
+        res = backend._lib.EC_KEY_set_private_key(ec_key, self.bn_key.bignum)
+        backend.openssl_assert(res == 1)
+
+        # Get public key
+        point = backend._lib.EC_POINT_new(self.bn_key.group)
+        backend.openssl_assert(point != backend._ffi.NULL)
+        point = backend._ffi.gc(point, backend._lib.EC_POINT_free)
+
+        with backend._tmp_bn_ctx() as bn_ctx:
+            res = backend._lib.EC_POINT_mul(
```

Comment:
We do point multiplication here with `NULL` - although I have no doubt this is correct, it's not obvious why it's correct.  What's actually happening here?

Reply:
From the OpenSSL docs on EC_POINT_mul, "EC_POINT_mul calculates the value generator * n + q * m and stores the result in r. The value n may be NULL in which case the result is just q * m."

So, it's only used to perform a multiplication without the other stuff.

Reply:
Ahh, got it.  Nice.